crates/wasi-common/yanix/src/filetime.rs

```diff
@@ -10,40 +10,39 @@ use std::io::Result;
 
 pub use super::sys::filetime::*;
 
-cfg_if::cfg_if! {
-    if #[cfg(not(target_os = "emscripten"))] {
-        fn filetime_to_timespec(ft: &filetime::FileTime) -> Result<libc::timespec> {
-            Ok(
-                libc::timespec {
-                    tv_sec: ft.seconds(),
-                    tv_nsec: i64::from(ft.nanoseconds()),
-                }
-            )
-        }
-    } else {
-        fn filetime_to_timespec(ft: &filetime::FileTime) -> Result<libc::timespec> {
-            use std::convert::TryInto;
-            use std::io::Error;
-            // Emscripten expects both `tv_sec` and `tv_nsec` fields to be `i32`.
-            // Here however `ft.seconds() -> i64` and `ft.nanoseconds() -> u32` so
-            // a simple `as` cast may be insufficient. So, perform a checked conversion,
-            // log error if any, and convert to libc::EOVERFLOW.
-            let tv_sec = match ft.seconds().try_into() {
-                Ok(sec) => sec,
-                Err(_) => {
-                    log::debug!("filetime_to_timespec failed converting seconds to required width");
-                    return Err(Error::from_raw_os_error(libc::EOVERFLOW));
-                }
-            };
-            let tv_nsec = match ft.nanoseconds().try_into() {
-                Ok(nsec) => nsec,
-                Err(_) => {
-                    log::debug!("filetime_to_timespec failed converting nanoseconds to required width");
-                    return Err(Error::from_raw_os_error(libc::EOVERFLOW));
-                }
-            };
-            Ok(libc::timespec { tv_sec, tv_nsec })
-        }
+pub(crate) trait FileTimeExt {
+    fn seconds_checked(&self) -> Result<libc::time_t>;
+    #[cfg(target_pointer_width = "32")]
+    fn nanoseconds_checked(&self) -> Result<i32>;
+    #[cfg(target_pointer_width = "64")]
+    fn nanoseconds_checked(&self) -> Result<i64>;
+}
+
+impl FileTimeExt for filetime::FileTime {
+    fn seconds_checked(&self) -> Result<libc::time_t> {
+        use std::convert::TryInto;
+        use std::io::Error;
+        let sec = match self.seconds().try_into() {
+            Ok(sec) => sec,
+            Err(_) => {
+                log::debug!("filetime_to_timespec failed converting seconds to required width");
+                return Err(Error::from_raw_os_error(libc::EOVERFLOW));
+            }
+        };
+        Ok(sec)
+    }
+    #[cfg(target_pointer_width = "32")]
+    fn nanoseconds_checked(&self) -> Result<i32> {
+        use std::convert::TryInto;
+        // According to [filetime] docs, since the nanoseconds value is always less than 1 billion,
+        // any value should be convertible to `i32`, hence we can `unwrap` outright.
+        //
+        // [filetime]: https://docs.rs/filetime/0.2.8/filetime/struct.FileTime.html#method.nanoseconds
+        Ok(self.nanoseconds().try_into().unwrap())
+    }
+    #[cfg(target_pointer_width = "64")]
```

Comment:
Could we avoid having this `#[cfg]`? It seems like it'd be best to just always have the `try_into().unwrap()`, and that I think means we could also remove  the `_checked` from the name, right?

Reply:
Agreed! You beat me to it actually! I was just reworking this bit as it seems silly to have `unwrap` wrapped in `Result` :-D